docs/_docs/internals/syntax.md

```diff
@@ -471,10 +471,16 @@ ConstrMods        ::=  {Annotation} [AccessModifier]
 ObjectDef         ::=  id [Template]                                            ModuleDef(mods, name, template)  // no constructor
 EnumDef           ::=  id ClassConstr InheritClauses EnumBody
 
-GivenDef          ::=  [GivenConditional '=>'] GivenSig
-GivenConditional  ::=  [DefTypeParamClause | UsingParamClause] {UsingParamClause}
-GivenSig          ::=  GivenType ['as' id] ([‘=’ Expr] | TemplateBody)
-                    |  ConstrApps ['as' id] TemplateBody
+GivenDef          ::=  [id ':'] GivenSig
+GivenSig          ::=  GivenImpl
+                    |  '(' ')' '=>' GivenImpl
+                    |  GivenConditional '=>' GivenSig
+GivenImpl         ::=  GivenType ([‘=’ Expr] | TemplateBody)
+                    |  ConstrApps TemplateBody
+GivenConditional  ::=  DefTypeParamClause
+                    |  DefTermParamClause
```

Comment:
is this really `DefTermParamClause`, or can it be `FunParamClause` instead? as far as i can tell, the difference is in modifiers (inline, erased) and default value.

Reply:
It s a DefTypeParamClause. Modifiers and default values are allowed.

Reply:
how do we resolve ambiguity between `DefTermParamClause` and ` '(' FunArgTypes ')' `, especially since a simple type can look like a [parameter](https://dotty.epfl.ch/docs/reference/experimental/named-tuples.html)?